modules/pipebackend/coprocess.cc

```diff
@@ -164,20 +164,11 @@ void CoProcess::receive(string &receive)
 {
   receive.clear();
     
-  if(d_timeout) {
-    struct timeval tv;
-    tv.tv_sec=d_timeout/1000;
-    tv.tv_usec=(d_timeout % 1000) * 1000;
-
-    fd_set rds;
-    FD_ZERO(&rds);
-    FD_SET(fileno(d_fp),&rds);
-    int ret=select(fileno(d_fp)+1,&rds,0,0,&tv);
-    if(ret<0)
-      throw PDNSException("Error waiting on data from coprocess: "+stringerror());
-    if(!ret)
-      throw PDNSException("Timeout waiting for data from coprocess");
-  }
+  int ret = waitForRWData(fileno(d_fp), true, 0, d_timeout*1000);
```

Comment:
Just a nit but I find the use of `waitForData()` easier to read (instead of having to look up what the second parameter to `waitForRWData()` does):

```suggestion
  int ret = waitForData(fileno(d_fp), 0, d_timeout*1000);
```

Reply:
Oh that is better, yes. Changed, tested, amended and force-pushed.